onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -388,12 +388,11 @@ ORT_API_STATUS_IMPL(OrtApis::RegisterCustomOpsLibrary, _Inout_ OrtSessionOptions
 }
 
 namespace {
-template <typename Loader>
-OrtStatus* CreateSessionImpl(_In_ const OrtEnv* env, _In_ const OrtSessionOptions* options,
-                             Loader loader, _Outptr_ OrtSession** out) {
+OrtStatus* CreateSessionImpl(_In_ const OrtEnv* /*env*/, _In_ const OrtSessionOptions* options,
+                             _In_ std::unique_ptr<::onnxruntime::InferenceSession>& sess,
+                             _Outptr_ OrtSession** out) {
```

Comment:
You're not changing any ownership here are you? If not, just provide InferenceSession& instead of the unique_ptr wrapped param.

Reply:
We are relinquishing ownership somewhere below - 

`*out = reinterpret_cast<OrtSession*>(sess.release());`

Reply:
Resolved per discussion.